spec/lib/search_spec.rb

```diff
@@ -2654,6 +2654,34 @@ def indexed_post(*args)
     end
   end
 
+  context "when plugin introduces a search_rank_sort_priorities modifier" do
+    before do
+      SearchIndexer.enable
+      DiscoursePluginRegistry.clear_modifiers!
+    end
+    after do
+      SearchIndexer.disable
+
+      DiscoursePluginRegistry.clear_modifiers!
+    end
+
+    it "allow modifying the search rank" do
+      plugin = Plugin::Instance.new
+      plugin.register_modifier(:search_rank_sort_priorities) do |ranks, search|
+        [["topics.closed", 77]]
+      end
```

Comment:
I'm thinking that the interface introduced by this API is too wide and the expected return value is not clear. It'll be nice if we can introduce something like this:

```
      plugin.register_modifier(:add_search_rank_sort_priorities) do |search|
        ["topics.closed", "77"]
      end
```

I don't think core's default ranks needs to be exposed in the API.

Reply:
I actually think it is fair to modify core. Allows a plugin to say .. "hey, I don't want a closed bump down" and we need to change nothing in core for that. not monkey patching ... etc...

Reply:
I guess that is fair too :+1: